Release/src/http/client/http_client_named_pipe.cpp

```diff
@@ -0,0 +1,466 @@
+            }
+            else
+            {
+                // While we won't be transfer-encoding the data, we will write it in portions.
+                named_pipe_context->m_bodyType = content_length_chunked;
+                named_pipe_context->m_remaining_to_write = content_length;
+            }
+        }
+
+        // TODO: Need better buffer management
+        std::ostringstream request_stream;
+        request_stream.imbue(std::locale::classic());
+        const auto &host = utility::conversions::to_utf8string(m_uri.host());
+
+        request_stream << utility::conversions::to_utf8string(msg.method()) << " " << utility::conversions::to_utf8string(encoded_resource) << " " << "HTTP/1.1" << CRLF;
+
+        // Add the Host header if user has not specified it explicitly
+        if (!msg.headers().has(header_names::host))
+        {
+            request_stream << "Host: " << host << CRLF;
+        }
+
+        // Add headers.
+        if (!msg.headers().empty())
+        {
+            request_stream << utility::conversions::to_utf8string(::web::http::details::flatten_http_headers(msg.headers()));
+        }
+
+        // TODO: Add support for "Connection: Keep-Alive" ?
+        //// Enforce HTTP connection keep alive (even for the old HTTP/1.0 protocol).
+        //request_stream << "Connection: Keep-Alive" << CRLF << CRLF;
+
+        // Add the body
+        auto rbuf = request->_get_readbuffer();
+
+        uint8_t*  block = nullptr;
+        size_t length = 0;
+        if (rbuf.acquire(block, length)) {
+            request_stream << CRLF << std::string(reinterpret_cast<char*>(block), length);
+        }
+
+        // Register for notification on cancellation to abort this request.
+        if(msg._cancellation_token() != pplx::cancellation_token::none())
+        {
+            // cancellation callback is unregistered when request is completed.
+            named_pipe_context->m_cancellationRegistration = msg._cancellation_token().register_callback([weak_named_pipe_context]()
+            {
+                auto lock = weak_named_pipe_context.lock();
+                if (!lock) return;
+                lock->cleanup();
+            });
+        }
+
+        // Call the callback function of user customized options.
+        try
+        {
+            client_config().invoke_nativehandle_options(m_namedPipe);
+        }
+        catch (...)
+        {
+            request->report_exception(std::current_exception());
+            return;
+        }
+
+
+        _start_request_send(named_pipe_context, content_length, request_stream);
+    }
+
+private:
+
+    void _start_request_send(const std::shared_ptr<named_pipe_request_context>& named_pipe_context, size_t content_length, std::ostringstream& request_stream)
+    {
+        DWORD bytesWritten = 0;
+
+        auto success = WriteFile(m_namedPipe, request_stream.str().c_str(), static_cast<DWORD>(request_stream.str().length()), &bytesWritten, nullptr);
```

Comment:
should you use FlushFileBuffers to make sure complete buffer is sent to the server?

Reply:
Actually I need to do that on the server side (https://msdn.microsoft.com/en-us/library/windows/desktop/aa365598(v=vs.85).aspx). Client is OK.

Thanks!